src/tx/builder/index.ts

```diff
@@ -0,0 +1,579 @@
+ * @param [options.denomination='aettos'] Denomination of amounts
+ * @throws {Error} Validation error
+ * @return Array with binary fields of transaction
+ */
+export function buildRawTx (
+  params: TxParamsCommon,
+  schema: TxField[],
+  { excludeKeys = [], denomination = AE_AMOUNT_FORMATS.AETTOS }: TxOptionsRaw = {}
+): Buffer[] {
+  const filteredSchema = schema.filter(([key]) => !excludeKeys.includes(key))
+
+  // Transform `amount` type fields to `aettos`
+  params = transformParams(params, filteredSchema, { denomination })
+  // Validation
+  const valid = validateParams(params, schema, { excludeKeys })
+  if (Object.keys(valid).length > 0) {
+    throw new InvalidTxParamsError('Transaction build error. ' + JSON.stringify(valid))
+  }
+
+  return filteredSchema
+    .map(([key, fieldType]: [keyof TxSchema, string, EncodingType]) => serializeField(
+      params[key], fieldType, params))
+}
+
+/**
+ * Unpack binary transaction
+ * @function
+ * @alias module:@aeternity/aepp-sdk/es/tx/builder
+ * @param binary Array with binary transaction field's
+ * @param schema Transaction schema
+ * @return Object with transaction field's
+ */
+export function unpackRawTx<Tx extends TxSchema> (
+  binary: Uint8Array | NestedUint8Array,
+  schema: TxField[]): RawTxObject<Tx> {
+  return schema
+    .reduce<any>(
+    (
+      acc,
+      [key, fieldType, prefix],
+      index
+    ) => Object.assign(acc, { [key]: deserializeField(binary[index], fieldType, prefix) }),
+    {}
+  )
+}
+
+export interface BuiltTx<Tx extends TxSchema, Prefix extends EncodingType> {
+  tx: EncodedData<Prefix>
+  rlpEncoded: Uint8Array
+  binary: Uint8Array | NestedUint8Array
+  txObject: Tx
+}
+
+export type TxParamsBuild = TxParamsCommon & {
+  denomination?: AeAmountFormats
+}
+/**
+ * Build transaction hash
+ * @function
+ * @alias module:@aeternity/aepp-sdk/es/tx/builder
+ * @param params Object with tx params
+ * @param type Transaction type
+ * @param [options={}] options
+ * @param [options.excludeKeys] Array of keys to exclude for validation and build
+ * @param [options.prefix] Prefix of transaction
+ * @throws {Error} Validation error
+ * @returns object
+ * @returns object.tx Base64Check transaction hash with 'tx_' prefix
+ * @returns object.rlpEncoded rlp encoded transaction
+ * @returns object.binary binary transaction
+ */
+export function buildTx<Prefix> (
+  params: TxParamsBuild,
+  type: TxType,
+  { excludeKeys = [], prefix = 'tx', vsn, denomination = AE_AMOUNT_FORMATS.AETTOS }: {
+    excludeKeys?: string[]
+    prefix?: EncodingType
+    vsn?: number
+    denomination?: AeAmountFormats
+  } = {}
+): Prefix extends EncodingType
+    ? BuiltTx<TxSchema, Prefix>
+    : BuiltTx<TxSchema, 'tx'> {
+  const schemas = TX_SERIALIZATION_SCHEMA[type]
+  if (schemas == null) throw new UnexpectedTsError()
+
+  vsn ??= Math.max(
+    ...Object.keys(schemas).map(a => +a))
+  const schema = schemas[vsn as keyof typeof TX_SERIALIZATION_SCHEMA[typeof type]]
```

Comment:
`as` part looks overcomplicated, isn't `van` just a number?

Reply:
It is, but schemas is an object of readonly keys. thereforce having simply as `schemas[vsn]` pops a tserror

Changing it to `vsn as keyof typeof schemas`
`